src/discounts/components/SaleValue/SaleValueTextField.tsx

```diff
@@ -0,0 +1,83 @@
+import { TextField } from "@material-ui/core";
+import { ChannelSaleData } from "@saleor/channels/utils";
+import { SaleType } from "@saleor/types/globalTypes";
+import React from "react";
+import { useEffect, useState } from "react";
+import { useIntl } from "react-intl";
+
+import { SaleValueInputChangeType } from "./types";
+
+interface SaleValueTextFieldProps {
+  dataType: keyof typeof SaleType;
+  helperText: string;
+  disabled: boolean;
+  listing: ChannelSaleData;
+  onChange: SaleValueInputChangeType;
+}
+
+const SaleValueTextField: React.FC<SaleValueTextFieldProps> = ({
+  dataType,
+  helperText,
+  disabled,
+  listing,
+  onChange
+}) => {
+  const intl = useIntl();
+
+  const [fixedValue, setFixedValue] = useState("");
+  const [percentageValue, setPercentageValue] = useState("");
+
+  const handleChange = (value: string) => {
+    onChange(listing.id, value);
+  };
+
+  const setCurrentValue = (value: string) => {
+    if (dataType === SaleType.PERCENTAGE) {
+      setPercentageValue(value);
+    } else {
+      setFixedValue(value);
+    }
+  };
+
+  useEffect(() => {
+    setCurrentValue(listing.discountValue);
+  }, []);
+
+  useEffect(() => {
```

Comment:
I think this should work the same and uses the logic we already have in getTextFieldValue?
```suggestion
  useEffect(() => handleChange(getTextFieldValue()), [dataType])
```